File: src/app/models/customer.ts
import { Address } from './address';
import { baseModel } from './baseModel';

export class Customer extends baseModel{
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    addresses: Address[];

    constructor() {
        super()
        this.id = 0;
        this.firstName = '';
        this.lastName = '';
        this.email = '';
        this.addresses = [
            new Address()
        ];
    }
}

File: api/Customers.WebApi/Models/Customer.cs
using Customers.WebApi;
using Customers.WebApi.Controllers;
using Customers.WebApi.Models;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomerDatabase.Shared
{
    public class Customer : BaseModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }

        public virtual List<Address> Addresses { get; set; }

    }
}
